stage0/src/kernel.rs

```diff
@@ -94,11 +90,11 @@ pub fn try_load_cmdline(fw_cfg: &mut FwCfg) -> Option<&'static CStr> {
         "cmdline size did not match expected size"
     );
 
-    let cmdline = CStr::from_bytes_with_nul(buf.leak()).expect("invalid kernel command-line");
-    log::debug!(
-        "Kernel cmdline: {}",
-        cmdline.to_str().expect("invalid kernel commande-line")
-    );
+    let cmdline = CString::from_vec_with_nul(buf)
```

Comment:
Maybe `CString::from_vec_with_nul(buf).and_then(|s| s.into_string()).expect("....")`, it'd mean we don't have to spell out the error twice.

Reply:
The compiler doesn't like that since the error types of the two results don't match.